docs: infrastructure (#10)
PredictiveMovement/predictivemovement: infra/README.md

  #### dev
  - for the `dev` environment we define everything in `k8s/base/stack`
  - the `kustomization.yaml` has a list of `resources` that are created, the namespace that will be used and a config map generator that creates [ConfigMaps](https://kubernetes.io/docs/concepts/configuration/configmap/) in our cluster (one for common properties reused by majority of pods and an engine specific one)

  #### prod
  - for the `prod` environment we define everything in `k8s/overlays/stack-prod`
  - the `kustomization.yaml` inside this reuses the base from `k8s/base/stack` and extends that with some customizations like [patchStrategicMerge](https://kubectl.docs.kubernetes.io/references/kustomize/glossary/#patchstrategicmerge) that allows us to duplicate less code (like the Ingress that should have a different URL between different environments), use a different namespace and define only different properties than the ones in `dev` with the [configMapGenerator](https://github.com/kubernetes-sigs/kustomize/blob/master/examples/configGeneration.md)

  #### k8s/base/dependencies and k8s/dependencies-prod

>   - this was a try at trying to use `kustomize` for separating the resource files for `dependencies` (minio, postgres, rabbitmq, redis...)

Author: ```suggestion
  - this was an attempt at trying to use `kustomize` for separating the resource files for `dependencies` (minio, postgres, rabbitmq, redis...)
```
